app/app/static/custom/ourscript.js

```diff
@@ -134,3 +264,54 @@ $('#librariansAvailableIncrement').click(function () {
 	let next = curr + 1;
 	document.getElementById('librariansAvailableCount').innerHTML = next;
 });
+
+window.onload = async (event) => {
+	console.info("Loading the Queue from API");
+
+	// Load every queue from the API parallel
+	const queueToLoad = Object.keys(queueList)
+
+	const requestPromises = queueToLoad.map(async (queue) => {
+		const response = await fetch("get_queue", {
+			method: "POST",
+			body: JSON.stringify({queue}),
+			headers: {
+				'Content-Type': 'application/json'
+		}})
+		const dataResponse = await response.json()
+		console.log
+		return dataResponse
+	})
+
+	const responsesResult = await Promise.all(requestPromises)
+	
+	// This assumes that queueToLoad is of the same index dimension as responsesResult (dimension by map)
+	queueToLoad.forEach((queue, index) => {
+		// Assigns the result of the response to the correct queueList state
+		queueList[queue] = responsesResult[index]["queue"]
+
+		// Set the timers correctly for each of the elements
+		responsesResult[index]["queue"].map(element => {
+
+			if (element.status == "In Queue") {
+				// Today - `enterQueueTime` is the relative date in seconds
+				const timeDifferenceMiliseconds = (new Date()).getTime() - (new Date(element.enterQueueTime)).getTime()
+
+				// Convert to seconds (rounded)
+				timers[element.id] = Math.round(timeDifferenceMiliseconds / 1000);
+				console.log(new Date())
+				console.log(element)
+				console.log(new Date(element.enterQueueTime))
+				console.log(timers[element.id])
+				timerIntervals[element.id] = setInterval(setTime, 1000, element.id);}
+
+			else {
+				const timeDifferenceMiliseconds = (new Date()).getTime() - (new Date(element.changeSessionTime)).getTime()
```

Comment:
Both if statement branch are exactly the same except for which element is being selected. Consider simplifying this by using a ternary operator on which element to select

Eg.
```javascript
const timeElement = element.status=="In Queue" ? "enterQueueTime" : "changeSessionTime"

//... Insert other code here
timeDifferenceMiliseconds = (new Date()). getTime() (new Date(element[timeElement] ) ). getTime()
//... Insert other code here
```

Reply:
Great idea! commited.